src/services/lease-service/routes/listings.ts

```diff
@@ -661,4 +662,76 @@ export const routes = (router: KoaRouter) => {
     ctx.status = 200
     ctx.body = { ...metadata }
   })
+
+  /**
+   * @swagger
+   * /listings/{listingId}/status:
+   *   put:
+   *     summary: Update a listings status by ID
+   *     description: Updates a listing status by it's ID.
+   *     tags:
+   *       - Listings
+   *     parameters:
+   *       - in: path
+   *         name: listingId
+   *         required: true
+   *         schema:
+   *           type: number
+   *         description: ID of the listing to delete.
+   *     requestBody:
+   *       required: true
+   *       content:
+   *          application/json:
+   *             schema:
+   *               type: object
+   *       properties:
+   *         status:
+   *           type: number
+   *           description: The listing status.
+   *     responses:
+   *       '200':
+   *         description: Successfully updated listing.
+   *       '404':
+   *         description: Listing not found.
+   *       '500':
+   *         description: Internal server error.
+   *         content:
+   *           application/json:
+   *             schema:
+   *               type: object
+   *               properties:
+   *                 error:
+   *                   type: string
+   *                   description: The error message.
+   */
+  const UpdateListingStatusRequestParamsSchema = z.object({
+    status: z.nativeEnum(ListingStatus),
+  })
+
+  router.put(
+    '(.*)/listings/:listingId/status',
+    parseRequestBody(UpdateListingStatusRequestParamsSchema),
+    async (ctx) => {
+      const metadata = generateRouteMetadata(ctx)
+      const result = await listingAdapter.updateListingStatuses(
+        [Number(ctx.params.listingId)],
+        ctx.request.body.status
+      )
+
+      if (!result.ok) {
+        if (result.err === 'no-update') {
+          ctx.status = 404
+          ctx.body = { reason: 'Listing not found', ...metadata }
```

Comment:
What about error codes? :) 

`listing-not-found`
`internal-error`

Reply:
https://github.com/Bostads-AB-Mimer/onecore-leasing/pull/157/commits/9d98277d5a05cf6748aa46d665559c2cc8c4c416